### PR TITLE
[Data] lock in data handle

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add mutex lock in data handle and update function description.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: src/libnnstreamer-edge/nnstreamer-edge-common.c [hunk 1]
```diff
@@ -645,9 +646,16 @@ nns_edge_data_destroy (nns_edge_data_h data_h)
   unsigned int i;
 
   ed = (nns_edge_data_s *) data_h;
+  if (!ed) {
+    nns_edge_loge ("Invalid param, given edge data handle is null.");
```
Comment (gichan-jang) [inline src/libnnstreamer-edge/nnstreamer-edge-common.c:650]: This comment is not directly related with PR.
If the user get this error message, It is still difficult for the user to know where the error occurred.
How about printing file name and line or function name?

**Review by gichan-jang — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]
LGTM
